sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -1184,3 +1210,38 @@ func withUnmaskedPaths(unmaskPaths []string) oci.SpecOpts {
 		return nil
 	}
 }
+
+// Convert label to map[string]string for containerd.WithPullLabels.
+// Label are in the format of "key=value".
+func convertLabels(labels string) (map[string]string, error) {
+	labelsMap := make(map[string]string)
+	labelsList := strings.Split(labels, ",")
+	for _, label := range labelsList {
+		labelKeyValue := strings.Split(label, "=")
+
+		if len(labelKeyValue) != 2 {
+			return labelsMap, fmt.Errorf("invalid label format: %s. It should be key=value", label)
+		}
```

Comment:
You could also tolerate input without an `=` and just assume it's a key where the value is the empty string.

Reply:
yeah, convertLabels function actually handle this situation and consider this is correct. I have unit test [here](https://github.com/bottlerocket-os/bottlerocket/pull/3757/files#diff-46e7ce9ff30b890e45ef5c58083d8fd181908851aa23c5c7db04d12bc5833ddcR283-R288)

Reply:
I think if we support like that, we might allow the host-ctr to accept incorrect label. If users specify the label like `value-key`. The label doesn't have `=`, and if we assume it's a key where the value is the empty string, I think this would misinterpret the user's request. We should strictly ask users to input like "key=value". For empty value, it could be `key=` and this would be accepted by host-ctr. Unit-test [here](https://github.com/bottlerocket-os/bottlerocket/pull/3757/files#diff-46e7ce9ff30b890e45ef5c58083d8fd181908851aa23c5c7db04d12bc5833ddcR283-R289)

Reply:
Is it valid to have the key be an empty string? If so, then I think that is probably a better approach. I wasn't sure if the containerd spec allows for that but we shouldn't enforce something more than what the upstream system can handle.

Reply:
https://github.com/opencontainers/image-spec/blob/365fa41472a19107b0b4cbc17e141a9b97f9b66e/annotations.md#rules the second point:
`While the value MUST be present, it MAY be an empty string.` so I think we should accept empty strings, my earlier comment was overly strict on what is valid.

Reply:
>While the value MUST be present, it MAY be an empty string.

Yeah, they support the value can be empty string. The reason why I think we need `=` here because containerD directly receives `map` and we need convert the slice of strings  to `map`.

Ok, I think I can allow the labels to accept the label format without an `=`.

source from [rules](https://github.com/opencontainers/image-spec/blob/365fa41472a19107b0b4cbc17e141a9b97f9b66e/annotations.md#rules).